### PR TITLE
Sign up I

### PR DESCRIPTION
**Pull request by raparicio6**
## Screenshots

image

## Trello Card

https://trello.com/c/CENfil6e/49-sign-up-i-layout

### CODE REVIEW

File: react/wbooks/src/app/components/SignUp/index.js [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+          <InputLabel
+            className=""
+            textClassName={`${styles.inputLabel} m-top-3`}
+            dataFor="name"
+            label={t('SignUp:name')}
+            inputClassName={`${styles.input} full-width`}
+            name="name"
+            inputId="name"
+            inputType="text"
+            handleChange={this.onChangeName}
```
Comment (erikankodelia) [inline react/wbooks/src/app/components/SignUp/index.js]: If the input label sends you the value and the name, you can use the same function for all the inputs in this form.

`onValueChange = (name, value) => this.setState({ [name]: value })`

or you can send the evt.target and do something like that:

`onValueChange = ({ id, value }) => this.setState({ [id]: value })`

And all the inputs are very similar, only change some props. So you can create an array with the props of the inputs and then use a map for render it. For example:

`FIELDS.map(props =>
<InputLabel
textClassName={'${styles.inputLabel} m-top-3'}
label={props.label}
.../>)`

File: react/wbooks/src/scss/components.scss [hunk 2]
```diff
@@ -1,7 +1,16 @@
+@mixin button {
```
[the remark this answers is not in the document]
Comment (rapariciof) [reply react/wbooks/src/scss/components.scss]: you mean to not use a mixin? Jota suggested me to use a mixin for the buttons so I continued with that approach

File: react/wbooks/src/scss/components.scss [hunk 3]
```diff
@@ -1,7 +1,16 @@
...
+  border-radius: 5px;
+  height: 30px;
+}
+
 .unlabeled-field {
   margin-top: 30px;
 }
 
+.full-height {
```
[the remark this answers is not in the document]
Comment (rapariciof) [reply react/wbooks/src/scss/components.scss]: not yet :D

File: react/wbooks/src/scss/layout.scss [hunk 4]
```diff
@@ -117,6 +117,7 @@
 }
 
 @for $i from 1 through 16 {
+
```
[the remark this answers is not in the document]
Comment (rapariciof) [reply react/wbooks/src/scss/layout.scss]: the style linter add it!

File: react/wbooks/src/scss/layout.scss [hunk 5]
```diff
@@ -125,6 +126,7 @@
...
```
[the remark this answers is not in the document]
Comment (rapariciof) [reply react/wbooks/src/scss/layout.scss]: the style linter add it!

File: react/wbooks/src/scss/layout.scss [hunk 6]
```diff
@@ -133,6 +135,7 @@
...
```
[the remark this answers is not in the document]
Comment (rapariciof) [reply react/wbooks/src/scss/layout.scss]: the style linter add it!

File: react/wbooks/src/scss/margins.scss [hunk 7]
```diff
@@ -34,6 +34,7 @@ $max-left: 50;
   $unscaled-max-top: $max-top / $scale;
 
   @for $i from $unscaled-min-top through $unscaled-max-top {
+
```
[the remark this answers is not in the document]
Comment (rapariciof) [reply react/wbooks/src/scss/margins.scss]: the style linter add it!

File: react/wbooks/src/scss/margins.scss [hunk 8]
```diff
@@ -47,6 +48,7 @@ $max-left: 50;
   $unscaled-max-right: $max-right / $scale;
 
   @for $i from $unscaled-min-right through $unscaled-max-right {
+
```
[the remark this answers is not in the document]
Comment (rapariciof) [reply react/wbooks/src/scss/margins.scss]: the style linter add it!

File: react/wbooks/src/scss/margins.scss [hunk 9]
```diff
@@ -60,6 +62,7 @@ $max-left: 50;
   $unscaled-max-bottom: $max-bottom / $scale;
 
   @for $i from $unscaled-min-bottom through $unscaled-max-bottom {
+
```
[the remark this answers is not in the document]
Comment (rapariciof) [reply react/wbooks/src/scss/margins.scss]: the style linter add it!

File: react/wbooks/src/scss/margins.scss [hunk 10]
```diff
@@ -73,6 +76,7 @@ $max-left: 50;
   $unscaled-max-left: $max-left / $scale;
 
   @for $i from $unscaled-min-left through $unscaled-max-left {
+
```
[the remark this answers is not in the document]
Comment (rapariciof) [reply react/wbooks/src/scss/margins.scss]: the style linter add it!

File: react/wbooks/src/app/components/InputLabel/index.js [hunk 11]
```diff
@@ -0,0 +1,75 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
```
[the remark this answers is not in the document]
Comment (raparicio6) [reply react/wbooks/src/app/components/InputLabel/index.js]: nice :)

File: react/wbooks/src/app/components/SignUp/styles.module.scss [hunk 12]
```diff
@@ -0,0 +1,45 @@
...
+}
+
+.input-label {
+  color: $black;
+  font-size: 15px;
+}
+
+.login-button {
+
+  @include button(); // stylelint-disable-line at-rule-no-unknown
```
[the remark this answers is not in the document]
Comment (raparicio6) [reply react/wbooks/src/app/components/SignUp/styles.module.scss]: can I use a mixin? Jota suggested me that before

File: react/wbooks/src/app/components/SignUp/index.js (hunk 1 above)
Comment (raparicio6) [reply react/wbooks/src/app/components/SignUp/index.js]: awesome :)

**Pull request merged**